PhysicsTools/NanoAOD/python/l1trig_cff.py

```diff
@@ -0,0 +1,189 @@
+import FWCore.ParameterSet.Config as cms
+from PhysicsTools.NanoAOD.nano_eras_cff import *
+from PhysicsTools.NanoAOD.common_cff import *
+
+l1PtVars = cms.PSet(
+    pt  = Var("pt",  float, precision=10),
+    phi = Var("phi", float, precision=10),
+)
+l1P3Vars = cms.PSet(
+    l1PtVars,
+    eta = Var("eta", float, precision=10),
+)
+
+l1ObjVars = cms.PSet(
+    l1P3Vars,
+    hwPt = Var("hwPt()",int,doc="hardware pt"),
+    hwEta = Var("hwEta()",int,doc="hardware eta"),
+    hwPhi = Var("hwPhi()",int,doc="hardware phi"),
+    hwQual = Var("hwQual()",int,doc="hardware qual"),
+    hwIso = Var("hwIso()",int,doc="hardware iso")
+)
+
+l1JetReducedVars = cms.PSet(
+    l1P3Vars
+)
+
+l1EtSumReducedVars = cms.PSet(
+    l1PtVars,
+    etSumType = Var("getType()",int,doc="et sum type"),
+)
+l1EGReducedVars = cms.PSet(
+    l1P3Vars,
+    hwIso = Var("hwIso()",int,doc="hardware iso")
+)
+
+l1TauReducedVars = cms.PSet(
+    l1P3Vars,
+    hwIso = Var("hwIso()",int,doc="hardware iso")
+)
+
+l1MuonReducedVars = cms.PSet(
+    l1P3Vars,
+    hwQual = Var("hwQual()",int,doc="hardware qual"),
+    hwCharge = Var("hwCharge()",int,doc=""), 
+    etaAtVtx = Var("etaAtVtx()",float,doc=""),
+    phiAtVtx = Var("phiAtVtx()",float,doc=""),
+    ptUnconstrained = Var("ptUnconstrained()",float,doc=""),
+    hwDXY = Var("hwDXY()",int,doc=""),
+)
+
+l1MuTable = cms.EDProducer("SimpleTriggerL1MuonFlatTableProducer",
+    src = cms.InputTag("gmtStage2Digis","Muon"),
+    minBX = cms.int32(-2),
+    maxBX = cms.int32(2),                           
+    cut = cms.string(""), 
+    name= cms.string("L1Mu"),
+    doc = cms.string(""),
+    extension = cms.bool(False), 
+    variables = cms.PSet(l1ObjVars,
+                         hwCharge = Var("hwCharge()",int,doc=""),
+                         hwChargeValid = Var("hwChargeValid()",int,doc=""),
+                         tfMuonIndex = Var("tfMuonIndex()",int,doc=""),
+                         hwTag = Var("hwTag()",int,doc=""),
+                         hwEtaAtVtx = Var("hwEtaAtVtx()",int,doc=""),
+                         hwPhiAtVtx = Var("hwPhiAtVtx()",int,doc=""),
+                         etaAtVtx = Var("etaAtVtx()",float,doc=""),
```

Comment:
There are still a few floating point variables stored with full precision, can you please have another look at reducing the precision?

Note that also integers can be stored signed or unsigned and with reduced precision (e.g. 8 or 16 bits).

Reply:
thanks, will do.

Reply:
FYI this was the cff with the datatypes I used for the size studies (I had modified it by e.g. putting more int16 and `precision=10`, but there are certainly ways to further reduce the precision)
https://gist.github.com/artlbv/d3e71fb9e4c37f5533bbffaecc6dfbcf

Reply:
I had a quick look at that but noted that there were still a few `float`s with full precision; also note that specifying the `precision` only works for floats, not ints (for integers you can only specify the type `[u]int[8|16|]`.

Reply:
ok thanks for checking!
Concerning the float precision: what does that actually mean? I found the code where this is defined but I think I still don't get it. Since the L1 float variables are actually all coming originally from HW values they all have a predetermined bit width. E.g. Jet Et is 12 bit (afaik). Does it mean we should set the float corresponding to that to `precision=12` ?

adding @eyigitba @lathomas

Reply:
All it does is zero out some bits in the mantissa and let the ROOT file compression take care of the zeros.

> Since the L1 float variables are actually all coming originally from HW values they all have a predetermined bit width

That's interesting, so in practice they already have limited precision (only a few non-zero bits in the mantissa). In that case setting the `precision` to any value not smaller than that will have no effect. The only thing to be decided is if you need the full precision coming from the hardware or if a smaller precision can be afforded (though we probably won't gain much by this).

We're planning to support ROOT fixed-range "floating point" storage in Nano, which works differently, but that's still WIP.

Reply:
let me also add @dinyar just in case I got something wrong :)